receiver/dcgmreceiver/scraper_gpu_test.go

```diff
@@ -198,24 +213,37 @@ func TestScrapeOnProfilingPaused(t *testing.T) {
 	scraper := newDcgmScraper(config, settings)
 	require.NotNil(t, scraper)
 
-	defer func() { testprofilepause.ResumeProfilingMetrics() }()
-	err := testprofilepause.PauseProfilingMetrics()
-	if err != nil {
-		if errors.Is(err, testprofilepause.FeatureNotSupportedError) {
-			t.Skipf("Pausing profiling not supported")
-		} else {
-			t.Errorf("Pausing profiling failed with error %v", err)
+	defer func() {
+		client, err := scraper.initClient()
+		require.NoError(t, err)
+		defer client.cleanup()
```

Comment:
I'm confused — why do we need this? Is it because we don't want to access `dcgm.handle` across threads? Or does that get overwritten somehow?

I actually suspect it might be enough to call `dcgmInit` (or even `dcgm.Init`) here:
```suggestion
		dcgmCleanup, err := dcgm.Init(dcgm.Standalone, config.TCPAddrConfig.Endpoint, "0")
		require.NoError(t, err)
		defer dcgmCleanup()
```

Reply:
This is needed because there is no active client at the time we're trying to pause/resume profiling.

You're right that a bare dcgm.Init/cleanup should be sufficient, but I was trying to avoid duplicating the knowledge of how to init here in the test. WDYT?

Reply:
(a) Then you shouldn't need that for resuming, only for pausing.
(b) Should we instead wait for the client to be created before trying to pause profiling?

Reply:
a) The resume happens in a defer, which means after `scraper.stop` had been called and the client has been uninitialized
b) No, because this is trying to test that fewer metrics are collected... if we delay pausing until after the client is created, it will see those metrics from before the pause.

Reply:
(a) Maybe this defer should happen in a smaller scope, before `scraper.stop()` is called?
(b) The test is probably just fragile. Doesn't the client collect new values since the last poll? Should we change it to return no points if no new values appeared since last scrape? Then this test will do the right thing (create scraper, trigger a scrape, pause profiling, trigger another scrape, check metrics from the last scrape, resume profiling, stop scraping).

Reply:
This test has always paused profiling, then started scraping, then checked metrics, then stopped scraping, then resumed profiling. 

Are you suggesting that the test should be more complicated? We talked briefly about that, but it's not clear to me what behavior we should actually be testing for in that case.

Reply:
I think I understand now that you don't like how complicated the test has become. I just moved the client initialization into `testprofilepause` - PTAL?

Reply:
This looks a lot cleaner and easier to read, thanks!